client/src/redux/prop-types.ts

```diff
@@ -347,3 +352,53 @@ export interface UserFetchState {
   errored: boolean;
   error: string | null;
 }
+
+// Exam Related Types:
+interface GeneratedExamAnswer {
+  id: string;
+  answer: string;
+}
+
+// Generated Exam (from API)
+export interface GeneratedExamQuestion {
+  id: string;
+  question: string;
+  answers: GeneratedExamAnswer[];
+}
+
+export interface GenerateExamResponse {
+  error?: string;
+  generatedExam?: GeneratedExamQuestion[];
+}
```

Comment:
Not for now, but for future use, I think we should help ourselves by using conditional types:

```typescript
type A = { error: string; result: null; } | { error: null; result: string };
```

Reply:
Sounds good, I'll add it to the list of post-mvp tasks